pkg/cmd/registry/rule/enable/enable.go

```diff
@@ -0,0 +1,177 @@
+package enable
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/registrycmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/rule/rulecmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/cmdutil"
+	"github.com/redhat-developer/app-services-cli/pkg/core/config"
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/core/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/core/logging"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/spf13/cobra"
+
+	registryinstanceclient "github.com/redhat-developer/app-services-sdk-go/registryinstance/apiv1internal/client"
+)
+
+type options struct {
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     logging.Logger
+	localizer  localize.Localizer
+	Context    context.Context
+
+	ruleType   string
+	config     string
+	registryID string
+
+	artifactID string
+	group      string
+}
+
+// NewEnableCommand creates a new command for enabling rule
+func NewEnableCommand(f *factory.Factory) *cobra.Command {
+
+	opts := &options{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		localizer:  f.Localizer,
+		Context:    f.Context,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "enable",
+		Short:   f.Localizer.MustLocalize("registry.rule.enable.cmd.description.short"),
+		Long:    f.Localizer.MustLocalize("registry.rule.enable.cmd.description.long"),
+		Example: f.Localizer.MustLocalize("registry.rule.enable.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) (err error) {
+
+			validator := rulecmdutil.Validator{
+				Localizer: opts.localizer,
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			err = validator.ValidateRuleType(opts.ruleType)
+			if err != nil {
+				return err
+			}
+
+			isValid, configs := validator.IsValidRuleConfig(opts.ruleType, opts.config)
+			if !isValid {
+				return opts.localizer.MustLocalizeError("registry.rule.common.error.invalidRuleConfig",
+					localize.NewEntry("RuleType", opts.ruleType),
+					localize.NewEntry("Config", opts.config),
+					localize.NewEntry("ValidConfigList", cmdutil.StringSliceToListStringWithQuotes(configs)),
+				)
+			}
+
+			instanceID, ok := cfg.GetServiceRegistryIdOk()
+			if !ok {
+				return opts.localizer.MustLocalizeError("artifact.cmd.common.error.noServiceRegistrySelected")
+			}
+
+			opts.registryID = instanceID
+
+			return runEnable(opts)
+		},
+	}
+
+	flags := registrycmdutil.NewFlagSet(cmd, f)
+	ruleFlags := rulecmdutil.NewFlagSet(cmd, f)
```

Comment:
If you embed the flag set from `registrycmdutil` into this instead of the root flag set then can avoid defining multiple flagsets here.

flagset > registryflagset > ruleflagset.